### DISCUSSION TITLE
Trouble with running LIS - adding Noah-3.6 sprinkler irrigation

### DISCUSSION THREAD

**Discussion by shaider2** [NASA-LIS/LISF#1584] [Q&A]
Hi,
I was trying to run the LIS. Unfortunately, it is stopping after few seconds. I have uploaded the slurm.out file, config and the log file for reference. It will be great if I can get some suggestion reargding this
[lislog.docx](https://github.com/user-attachments/files/16366373/lislog.docx)
[lis.config.docx](https://github.com/user-attachments/files/16366383/lis.config.docx)
<img width="490" alt="slurm_out" src="https://github.com/user-attachments/assets/46cf2758-31aa-4e0c-89e1-e66317490ae7">

**Comment by dmocko** [comment]
Hi @shaider2 

We have not enabled the coupler between Noah-3.6 LSM and the sprinkler irrigation routine.

Looking in this routine: https://github.com/NASA-LIS/LISF/blob/master/lis/plugins/LIS_lsmirrigation_pluginMod.F90

You will see that only the Noah-3.3, CLSM-F2.5, RUC-3.7, Noah-MP-3.6, and Noah-MP-4.0.1 LSMs currently have
sprinkler irrigation enabled in LIS:

```
LIS_lsmirrigation_pluginMod.F90:        trim(LIS_sprinklerIrrigationId)//char(0),noah33_getirrigationstates)
LIS_lsmirrigation_pluginMod.F90:        trim(LIS_sprinklerIrrigationId)//char(0),clsmf25_getirrigationstates)
LIS_lsmirrigation_pluginMod.F90:        trim(LIS_sprinklerIrrigationId)//char(0),ruc37_getirrigationstates)
LIS_lsmirrigation_pluginMod.F90:        trim(LIS_sprinklerIrrigationId)//char(0),NoahMP36_getirrigationstates)
LIS_lsmirrigation_pluginMod.F90:        trim(LIS_sprinklerIrrigationId)//char(0),NoahMP401_getirrigationstates)
```

If you want to use Noah-3.6 instead, you can add/edit code similar to the Noah-3.3 plugins as well as in a directory
similar to this: https://github.com/NASA-LIS/LISF/tree/master/lis/surfacemodels/land/noah.3.3/irrigation

One other file to add/edit irrigation for Noah-3.6 LSM:  https://github.com/NASA-LIS/LISF/blob/master/lis/make/default.cfg

**Reply by shaider2** [reply]
Hi,
Thank you so much for the information. I was trying to add the noah33_getirrigationstates.F90 file in (LIS/LISF/tree/master/lis/surfacemodels/land/noah.3.6/irrigation) folder. However, I was still getting the same error. Is there any instructions for incorporating this plugin?
Thank you so much again for your support!

**Reply by dmocko** [reply]
Hi @shaider2

The actual path in your directory on your system (not on GitHub) should be:
    LISF/lis/surfacemodels/land/noah.3.6/irrigation

It should have noah36_getirrigationstates.F90 in it, where in the routine, replace all "noah33" with "noah36".

Add this at the end of the [Noah.3.6] section in your default.cfg:
`virtual_irrigation path: surfacemodels/land/noah.3.6/irrigation`

Did you add this line to LIS_lsmirrigation_pluginMod.F90 in LISF/lis/plugins ?:
`trim(LIS_sprinklerIrrigationId)//char(0),noah36_getirrigationstates)`

After you made your code updates, please be sure to go into LISF/lis/make, type "gmake realclean",
and then ./configure and ./compile as before.

**Comment by shaider2** [comment]
Hello @dmocko,
Thank you so much for the suggestions. I was able to complete all the steps you suggested. However, when I tried to compile the LIS again, it failed. It will be really kind if you can provide some feedback regarding this issue. thank you so much for your time and support.

**Reply by dmocko** [reply]
HI @shaider2 

Please attach a screenshot or the text showing the error of compilation.

**Reply by shaider2** [reply]
Dear @dmocko ,
Thank you for the response. I have added the error file and slurm.out file for you perusal.Please let me know if I need to provide any further information.
Thank you again for your support!
[job.7258251.err.docx](https://github.com/user-attachments/files/16806126/job.7258251.err.docx)
[slurm-7258251.out.docx](https://github.com/user-attachments/files/16806127/slurm-7258251.out.docx)

**Comment by dmocko** [comment]
Hi @shaider2 

Thanks for your patience.  I have been busy with meetings and other urgent tasks, and am only able to return to this now.

Can you please share your entire branch, so I can look at all of your code (after you have committed all of your changes)?

If that's not possible, at a minimum please share these files:
1. LISF/lis/make/default.cfg
2. LISF/lis/plugins/LIS_lsmirrigation_pluginMod.F90
3. LISF/lis/surfacemodels/land/noah.3.6/irrigation/noah36_getirrigationstates.F90
